pkg/controller/worker/machines_test.go

```diff
@@ -818,6 +861,29 @@ func expectGetSecretCallToWork(c *mockclient.MockClient, awsAccessKeyID, awsSecr
 		})
 }
 
+// expectDeleteAllCallToWork is the intialization methods which mocks the
+// delete call to delete all awsMachineClass from awsMachineClassList
+func expectDeleteAllCallToWork(c *mockclient.MockClient, awsMachineClassList *machinev1alpha1.AWSMachineClassList) {
+	c.EXPECT().
+		DeleteAllOf(context.TODO(), gomock.AssignableToTypeOf(&machinev1alpha1.AWSMachineClass{}), gomock.Any()).
+		AnyTimes().Do(func(_ context.Context, awsMachineClass *machinev1alpha1.AWSMachineClass, namespace client.InNamespace) error {
```

Comment:
Why AnyTimes()?

```suggestion
		.Do(func(_ context.Context, awsMachineClass *machinev1alpha1.AWSMachineClass, namespace client.InNamespace) error {
```

Reply:
Removed anytimes now. Was a carry over from the GCP where i needed it :( . Refer - https://github.com/gardener/gardener-extension-provider-aws/pull/212/files#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R928